gateway/handler.go

```diff
@@ -68,7 +69,83 @@ type handler struct {
 //
 // [IPFS HTTP Gateway]: https://specs.ipfs.tech/http-gateways/
 func NewHandler(c Config, backend IPFSBackend) http.Handler {
-	return newHandlerWithMetrics(&c, backend)
+	handler := newHandlerWithMetrics(&c, backend)
+
+	// Use the configured timeout or fall back to a default value
+	timeout := c.ResponseWriteTimeout
+	if timeout == 0 {
+		timeout = 30 * time.Second // Default timeout of 30 seconds
+	}
+
+	// Apply the timeout middleware
+	return WithResponseWriteTimeout(handler, timeout)
```

Comment:
Hm.. this makes disabling timeout impossible. 

Let's change this to no timeout by default (Kubo and Rainbow will set the default themselves), and skip entire things if no timeout.

```suggestion
	if c.ResponseWriteTimeout != 0 {
		handler = WithResponseWriteTimeout(handler, timeout)
	}

	// Apply the timeout middleware
	return handler
```